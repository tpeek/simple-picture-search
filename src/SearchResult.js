import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import ThumbUp from '@material-ui/icons/ThumbUp';
import Star from '@material-ui/icons/StarRounded';

const savedMessageStyles = {
  color: 'white',
  position: 'absolute',
  textAlign: 'center',
  bottom: -16,
  width: 200,
  padding: '2px 0',
};

const useStyles = makeStyles({
  root: {
    position: 'relative',
    maxWidth: 400,
    margin: '40px 0',
    '&::after': {
      content: '" "',
      clear: 'both',
      display: 'table',
    },
  },
  img: {
    float: 'left',
    width: 200,
    '&:hover': {
      '& + p': {
        display: 'block',
      },
    },
  },
  saveMessage: {
    ...savedMessageStyles,
    display: 'none',
    backgroundColor: '#be4362',
  },
  savedMessage: {
    ...savedMessageStyles,
    backgroundColor: '#ea8928',
  },
  tagContainer: {
    float: 'left',
    display: 'flex',
    flexWrap: 'wrap',
    width: 160,
    marginLeft: 16,
  },
  tag: {
    color: 'white',
    backgroundColor: '#00a48f',
    padding: '2px 5px',
    margin: 4,
    display: 'inline-block',
    fontSize: 14,
  },
  socialContainer: {
    position: 'absolute',
    bottom: 0,
    left: 224,
    fontSize: 12,
    fontWeight: 'bold',
  },
  thumb: {
    marginLeft: 5,
    marginBottom: -2,
    height: 14,
    width: 14,
    marginRight: 32,
  },
  star: {
    marginLeft: 5,
    marginBottom: -3,
    height: 16,
    width: 16,
  },
});

const SearchResult = ({ previewURL, tags, likes, favorites, saveResult, saved }) => {
  const classes = useStyles();

  return (
    <li className={classes.root}>
      <img src={previewURL} className={classes.img} onClick={saveResult} alt={tags} />
      <p className={saved ? classes.savedMessage : classes.saveMessage}>
        {saved ? 'Saved' : 'Save'}
      </p>
      <div className={classes.tagContainer}>
        {tags.split(', ').map((category) => (
          <span className={classes.tag} key={category}>
            {category}
          </span>
        ))}
      </div>
      <div className={classes.socialContainer}>
        <span>{likes}</span>
        <ThumbUp className={classes.thumb} />
        <span>{favorites}</span>
        <Star className={classes.star} />
      </div>
    </li>
  );
};

export default SearchResult;
