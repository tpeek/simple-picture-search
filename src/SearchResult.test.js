import React from 'react';
import SearchResult from './SearchResult';

const defaultProps = {
  previewURL: 'https://www.example.com',
  tags: 'animals, tigers, circus',
  likes: 9001,
  favorites: 42,
  saveResult: jest.fn(),
  saved: false,
};

const setup = (props) => shallow(<SearchResult {...defaultProps} {...props} />);

describe('<SearchResult />', () => {
  it('renders snapshot correctly', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('shows "Saved" text when saved = true', () => {
    const wrapper = setup({ saved: true });

    expect(wrapper.find('p').text()).toBe('Saved');
  });

  it('sets proper class when saved = true', () => {
    const wrapper = setup({ saved: true });

    expect(wrapper.find('p').props().className).toBe('makeStyles-savedMessage-4');
  });

  it('shows "Save" text when saved = false', () => {
    const wrapper = setup({ saved: false });

    expect(wrapper.find('p').text()).toBe('Save');
  });

  it('sets proper class when saved = false', () => {
    const wrapper = setup({ saved: false });

    expect(wrapper.find('p').props().className).toBe('makeStyles-saveMessage-3');
  });
});
