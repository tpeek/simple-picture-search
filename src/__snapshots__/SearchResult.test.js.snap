// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResult /> renders snapshot correctly 1`] = `
<li
  className="makeStyles-root-1"
>
  <img
    alt="animals, tigers, circus"
    className="makeStyles-img-2"
    onClick={[MockFunction]}
    src="https://www.example.com"
  />
  <p
    className="makeStyles-saveMessage-3"
  >
    Save
  </p>
  <div
    className="makeStyles-tagContainer-5"
  >
    <span
      className="makeStyles-tag-6"
      key="animals"
    >
      animals
    </span>
    <span
      className="makeStyles-tag-6"
      key="tigers"
    >
      tigers
    </span>
    <span
      className="makeStyles-tag-6"
      key="circus"
    >
      circus
    </span>
  </div>
  <div
    className="makeStyles-socialContainer-7"
  >
    <span>
      9001
    </span>
    <ThumbUpIcon
      className="makeStyles-thumb-8"
    />
    <span>
      42
    </span>
    <StarRoundedIcon
      className="makeStyles-star-9"
    />
  </div>
</li>
`;
